docker/tango/ska-python-buildenv/Pipfile

```diff
@@ -4,7 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-numpy = "==1.15.4"
+numpy = "==1.17.2"
```

Comment:
When creating the original images a year ago, I had a problem using 'numpy="*"' when using the base images as a user: whenever a new version of numpy was released, the mismatch between the 'latest' version and the cached version inside the -buildenv image caused the precompiled PyTango to be invalidated, resulting in PyTango being recompiled on every 'make build' and on every CI run. I got around this by pinning the versions in the user project to exactly the same version used in the -buildenv, which is easiest when the buildenv version is also fixed.

I didn't explore the PyTango build mechanics in detail, but I did see that Numpy version is detected and referenced as a compiler flag in the PyTango build procedure (e.g., -D PYTANGO_NUMPY_VERSION=1.17.2); maybe that's what caused the cached version to be ignored when Numpy updated? Either way, it could be worth verifying if PyTango is still recompiled in the user images when the numpy version differs from the base image. Ideally we'd be able to say 'numpy="*"' in both base images and user projects, but I'm not sure that's possible without effort to keep base images in sync whenever a new version of NumPy is released.

Reply:
Oh! Importantly, projects downstream should also use 1.17.2! If we don't want to require changes to downstream Pipfiles just yet, maybe we can pin NumPy to the old v1.15.4?

Reply:
That is the main problem with this approach. We cannot ask people to change every time their pipfile... I think the best is to compile pytango so that we avoid all the problems we had till now.

Reply:
I agree that, ideally, projects downstream would inherit the version of PyTango and Numpy from base images, whatever version they may be. I think there are two ways to approach this: we can experiment with downstream projects specifying 'numpy = ">= 1.1"' etc. to see if they inherit the base version, and/or we can lock down the pipfile repositories so that only SKA Nexus is used. On this last point, it would be *great* if SKA Nexus could start hosting PyTango wheels, then most of the -buildenv and PyTango cache complexity is moot: it doesn't matter if a project requires PyTango, it just downloads the precompiled wheel instead of recompiling.

Reply:
I see your point. I will talk about it with the team...